### PR TITLE
Add Benchmark output

### PR DESCRIPTION
**Pull request by mkarunan**
Log Output for contraction, permutation and reduction tests.

### CODE REVIEW

File: test/01_contraction/contraction_test.cpp [hunk 1]
```diff
@@ -727,6 +864,11 @@ namespace hiptensor
                 tolerance += epsilon * 2;
             }
 
+            size_t elementsCD = std::accumulate(d_ms_ns.mLengths.begin(),
```
Comment (cgmillette) [inline test/01_contraction/contraction_test.cpp]: needed?

File: test/02_permutation/permutation_test.cpp [hunk 2]
```diff
@@ -260,6 +337,34 @@ namespace hiptensor
...
+                                           hipDataTypeSize(abDataType),
+                                           std::multiplies<size_t>());
+
+            size_t sizeB = std::accumulate(extentB.begin(),
+                                           extentB.end(),
+                                           hipDataTypeSize(abDataType),
+                                           std::multiplies<size_t>());
+
+            mElapsedTimeMs        = float64_t(timeMs);
+            mTotalGFlops          = 2.0 * ((sizeA * sizeB) / hipDataTypeSize(abDataType));
```
Comment (cgmillette) [inline test/02_permutation/permutation_test.cpp:348]: For permute, the amount of data would be 2*sizeA, once for reading the tensor and once for writing.

File: test/03_reduction/reduction_test.cpp [hunk 3]
```diff
@@ -359,6 +436,33 @@ namespace hiptensor
...
+                                           hipDataTypeSize(acDataType),
+                                           std::multiplies<size_t>());
+
+            size_t sizeCD = std::accumulate(extentC.begin(),
+                                            extentC.end(),
+                                            hipDataTypeSize(acDataType),
+                                            std::multiplies<size_t>());
+
+            mElapsedTimeMs        = float64_t(timeMs);
+            mTotalGFlops          = 2.0 * ((sizeA * sizeCD) / hipDataTypeSize(acDataType));
```
Comment (cgmillette) [inline test/03_reduction/reduction_test.cpp:456]: Size for reduction is sizeA + sizeC + (sizeC if beta !=0)

File: test/02_permutation/permutation_test.cpp (hunk 2 above)
Comment (mkarunan) [reply test/02_permutation/permutation_test.cpp:348]: Yes Chris, above is GFlops(no of operations) , bytes noted below (sizeA + sizeB)

File: test/03_reduction/reduction_test.cpp (hunk 3 above)
Comment (mkarunan) [reply test/03_reduction/reduction_test.cpp:456]: same for reduction , 2 * sizeA * sizeCD - No of operations
down below , bytes = sizeA + sizeCD + (beta != 0 ? sizeCD : 0)

**Review by cgmillette — APPROVED** [review]
LGTM!

**Comment by mkarunan** [comment]
waiting for CI to pass, Assigning to @cgmillette

**Pull request merged**